node/src/actors/chain_manager/superblock.rs

```diff
@@ -187,23 +192,30 @@ pub fn mining_build_superblock(
         .map(|b| b.merkle_roots.tally_hash_merkle_root)
         .collect();
 
-    let pkh_hashes: Vec<Hash> = sorted_ars_identities.iter().map(|pkh| pkh.hash()).collect();
-
     Some(SuperBlock {
+        ars_length: ars_ordered_hash_leaves.len() as u64,
         data_request_root: hash_merkle_tree_root(&merkle_drs),
         tally_root: hash_merkle_tree_root(&merkle_tallies),
-        ars_root: hash_merkle_tree_root(&pkh_hashes),
+        ars_root: hash_merkle_tree_root(ars_ordered_hash_leaves),
         index,
         last_block,
         last_block_in_previous_superblock,
     })
 }
 
+/// Takes a set of keys and calculates their hashes roots to be used as leaves.
+pub fn hash_key_leaves(ars_ordered_keys: &[Bn256PublicKey]) -> Vec<Hash> {
+    ars_ordered_keys
+        .iter()
+        .map(|bn256| Hash::SHA256(calculate_sha256(&bn256.clone().to_uncompressed().unwrap()).0))
```

Comment:
Do you think we should move this logic to `impl Hashable for Bn256PublicKey`? In other words, do we want to always hash the uncompressed public key, or do we want to sometimes also hash the compressed public key? I would prefer to always hash the uncompressed so we don't run into errors, but hashing the compressed key should be faster (because we don't have to uncompress it).

Reply:
what we could run into is that we need to hash both, and create a merkle root with both. But in both cases I think we benefit  from moving it to hashable